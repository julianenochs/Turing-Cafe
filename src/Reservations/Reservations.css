.reservation__card {
    background-color: white;
    border: 2px solid black;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
}