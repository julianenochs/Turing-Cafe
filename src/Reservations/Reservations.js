import React from 'react';
import './Reservations.css';

const Reservations = ({key, id, name, date, time, number, cancelReservation}) => {
    return(
        <section className='reservation__card'>
            <h1>{name}</h1>
            <p>{date}</p>
            <p>{time} pm</p>
            <p>Number of guests: {number}</p>
            <button className='cancel-reservation' onClick={cancelReservation(id)}>Cancel</button>
        </section>
    )
}
export default Reservations;