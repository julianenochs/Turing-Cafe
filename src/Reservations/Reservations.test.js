import React from 'react';
import { shallow } from 'enzyme';
import Reservations from './Reservations';

describe('Reservations', () => {
    let wrapper;
    let mockCancelReservation = jest.fn()
    beforeEach(() => {
        wrapper = shallow(<Reservations 
                    key={3}
                    id={3}
                    name={'Julian'}
                    date={'2019/10/14'}
                    time={'11:00pm'}
                    number={3}
                    cancelReservation={mockCancelReservation}
                    />)
    });

    it('should match the snapshot with all of the data passed through', () => {
        expect(wrapper).toMatchSnapshot();
    });
});