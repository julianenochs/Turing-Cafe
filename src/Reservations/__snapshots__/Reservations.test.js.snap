// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservations should match the snapshot with all of the data passed through 1`] = `
<section
  className="reservation__card"
>
  <h1>
    Julian
  </h1>
  <p>
    2019/10/14
  </p>
  <p>
    11:00pm
     pm
  </p>
  <p>
    Number of guests: 
    3
  </p>
  <button
    className="cancel-reservation"
  >
    Cancel
  </button>
</section>
`;
